extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -29,11 +45,49 @@
 
     @BuildStep(onlyIf = IsDevelopment.class)
     @Record(ExecutionTime.STATIC_INIT)
-    public DevConsoleRuntimeTemplateInfoBuildItem collectBeanInfo(ArcRecorder recorder) {
+    public DevConsoleRuntimeTemplateInfoBuildItem exposeArcContainer(ArcRecorder recorder) {
         return new DevConsoleRuntimeTemplateInfoBuildItem("arcContainer",
                 new ArcContainerSupplier());
     }
 
+    @BuildStep(onlyIf = IsDevelopment.class)
+    void monitor(ArcConfig config, BuildProducer<DevConsoleRuntimeTemplateInfoBuildItem> runtimeInfos,
+            BuildProducer<AdditionalBeanBuildItem> beans, BuildProducer<AnnotationsTransformerBuildItem> annotationTransformers,
+            CustomScopeAnnotationsBuildItem customScopes,
+            List<BeanDefiningAnnotationBuildItem> beanDefiningAnnotations) {
+        if (!config.devMode.monitoringEnabled) {
+            return;
+        }
+        // Events
+        runtimeInfos.produce(new DevConsoleRuntimeTemplateInfoBuildItem("eventsMonitor",
+                new BeanLookupSupplier(EventsMonitor.class)));
+        beans.produce(AdditionalBeanBuildItem.unremovableOf(EventsMonitor.class));
+        // Invocations
+        beans.produce(AdditionalBeanBuildItem.builder().setUnremovable()
+                .addBeanClasses(InvocationTree.class, InvocationsMonitor.class, InvocationInterceptor.class, Monitored.class)
+                .build());
+        Set<DotName> skipNames = new HashSet<>();
+        skipNames.add(DotName.createSimple(InvocationTree.class.getName()));
+        skipNames.add(DotName.createSimple(InvocationsMonitor.class.getName()));
+        skipNames.add(DotName.createSimple(EventsMonitor.class.getName()));
+        // TODO skip all fwk beans?
```

Comment:
Personally I'd keep them as it may be valuable in getting a complete picture of what's happening.

Alternatively, there could be yet another config option that would filter them out?

Reply:
I'll just remove this TODO... I don't think we need more config options now ;-)